### PR TITLE
Improve code quality

### PR DESCRIPTION
**Pull request by sukhwinder33445**
Fixes php 8.2 relevant errors (found by phpstan and psalm) and removes dead code

### CODE REVIEW

File: library/Businessprocess/Modification/NodeAction.php [hunk 1]
```diff
@@ -110,6 +109,7 @@ protected function error($msg)
      */
     public static function create($actionName, $nodeName)
     {
+        /** @var static $className */
```
Comment (nilmerg) [inline library/Businessprocess/Modification/NodeAction.php:112]: $object is `static`, $className is `string`

File: library/Businessprocess/Modification/NodeAction.php [hunk 2]
```diff
@@ -144,7 +144,7 @@ public function serialize()
      */
     public static function unSerialize($string)
     {
-        $object = json_decode($string, JSON_FORCE_OBJECT);
+        $object = json_decode($string, true);
```
Comment (nilmerg) [inline library/Businessprocess/Modification/NodeAction.php:147]: should be false, I guess?

File: library/Businessprocess/Web/Form/Validator/HostServiceTermValidator.php [hunk 3]
```diff
@@ -60,10 +60,12 @@ public function isValid($terms)
 
             if ($this->parent->hasChild($term->getSearchValue())) {
                 $term->setMessage($this->translate('Already defined in this process'));
-            } else {
-                $testConfig->getNode('__unbound__')
-                    ->addChild($node);
+
+                return false;
```
Comment (nilmerg) [inline library/Businessprocess/Web/Form/Validator/HostServiceTermValidator.php:64]: And what about the other terms? Shouldn't they be still validated as well? (They should)

**Review by nilmerg — CHANGES_REQUESTED** [review]

File: library/Businessprocess/Modification/NodeAction.php (hunk 2 above)
Comment (sukhwinder33445) [reply library/Businessprocess/Modification/NodeAction.php:147]: JSON_FORCE_OBJECT = `16`, in bool `true`

**Review by nilmerg — DISMISSED** [review]

**Review by nilmerg — COMMENTED** [review]
Please rebase and tidy up your commits as described today.

**Comment by sukhwinder33445** [comment]
Fixes moved to #378.

**Pull request closed without merging**